### ISSUE TITLE
run_demo.bash error

### ISSUE THREAD

**Issue by tanakataiki2** [osrf/car_demo#66]
I installed rocker via pip and succesfuly build but following error occurs does anybody know how to solve this?
This issue is probablly because docker is incorrect but error message was different from other issue so I made this issue

:~/kinetic_ws/src/car_demo$ ./run_demo.bashCONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
/home/taiki/.local/lib/python2.7/site-packages/requests/__init__.py:83: RequestsDependencyWarning: Old version of cryptography ([1, 2, 3]) may cause slowdown.
  warnings.warn(warning, RequestsDependencyWarning)
Plugins found: ['dev_helpers', 'home', 'nvidia', 'pulse', 'user', 'x11']
Active extensions ['nvidia', 'x11']
Traceback (most recent call last):
  File "/home/taiki/.local/bin/rocker", line 11, in <module>
    sys.exit(main())
  File "/home/taiki/.local/lib/python2.7/site-packages/rocker/cli.py", line 58, in main
    dig = DockerImageGenerator(active_extensions, args_dict, base_image)
  File "/home/taiki/.local/lib/python2.7/site-packages/rocker/core.py", line 77, in __init__
    self.dockerfile = generate_dockerfile(active_extensions, self.cliargs, base_image)
  File "/home/taiki/.local/lib/python2.7/site-packages/rocker/core.py", line 180, in generate_dockerfile
    dockerfile_str += el.get_preamble(args_dict) + '\n'
  File "/home/taiki/.local/lib/python2.7/site-packages/rocker/nvidia_extension.py", line 107, in get_preamble
    return em.expand(preamble, self.get_environment_subs(cliargs))
  File "/home/taiki/.local/lib/python2.7/site-packages/rocker/nvidia_extension.py", line 92, in get_environment_subs
    dist, ver, codename = detect_os(cliargs['base_image'])
TypeError: 'NoneType' object is not iterable

Thanks

**Comment by mjbogusz** [comment]
Same error here.

Error log (same as above just formatted):
```sh
$ ./run_demo.bash 
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
Plugins found: ['dev_helpers', 'home', 'nvidia', 'pulse', 'user', 'x11']
Active extensions ['nvidia', 'x11']
Traceback (most recent call last):
  File "/usr/bin/rocker", line 11, in <module>
    load_entry_point('rocker==0.1.4', 'console_scripts', 'rocker')()
  File "/usr/lib/python3/dist-packages/rocker/cli.py", line 58, in main
    dig = DockerImageGenerator(active_extensions, args_dict, base_image)
  File "/usr/lib/python3/dist-packages/rocker/core.py", line 77, in __init__
    self.dockerfile = generate_dockerfile(active_extensions, self.cliargs, base_image)
  File "/usr/lib/python3/dist-packages/rocker/core.py", line 180, in generate_dockerfile
    dockerfile_str += el.get_preamble(args_dict) + '\n'
  File "/usr/lib/python3/dist-packages/rocker/nvidia_extension.py", line 107, in get_preamble
    return em.expand(preamble, self.get_environment_subs(cliargs))
  File "/usr/lib/python3/dist-packages/rocker/nvidia_extension.py", line 92, in get_environment_subs
    dist, ver, codename = detect_os(cliargs['base_image'])
TypeError: 'NoneType' object is not iterable
```

Versions:
```sh
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.3 LTS
Release:	18.04
Codename:	bionic

$ docker --version
Docker version 18.09.7, build 2d0083d
```

I know running this on 18.04 is not _supported_, but considering its in docker, it _should_ work the same.
Let me know if any additional software versions may be relevant.

**Comment by tfoote** [comment]
These look like instances of https://github.com/osrf/rocker/issues/43


I'm running a pretty similar setup, rocker installed from debian packages.
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.2 LTS
Release:	18.04
Codename:	bionic
$ docker --version
Docker version 18.09.7, build 2d0083d
```
However I can't reproduce it with the hosted docker image `osrf/car_demo` or rebuilding the image locally.  Can you try verifying it using the image `osrf/car_demo` that's online?

**Comment by tanakataiki2** [comment]
@tfoote 
Thanks you for your reply 

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.6 LTS
Release:	16.04
Codename:	xenial
$ docker --version
Docker version 19.03.1, build 74b1e89e8a

1. What does osrf/car_demo that's online mean?
I am new to docker so, could you tell me what the docker entry point is?

2. Is there any steps required?
I just did ./build_demo.bash I thought nothing else is required other than that.
I might have skipped some of these things 
    An X server
    Docker
    nvidia-docker2 
    The current user is a member of the docker group or other group with docker execution rights.
    rocker : I installed from pip and deb package


3. What does docker execution rights mean?

**Comment by mjbogusz** [comment]
```sh
docker rmi *
docker pull osrf/car_demo
rocker --nvidia --x11 --devices /dev/input/js0 -- osrf/car_demo
```
Same results.

Rocker version:
```sh
$ apt show python3-rocker
Package: python3-rocker
Version: 0.1.4-100
...
APT-Manual-Installed: yes
APT-Sources: http://packages.ros.org/ros/ubuntu bionic/main amd64 Packages
```

**Comment by czhherry** [comment]
I had the same problem and solved it by doing as this pull request https://github.com/osrf/rocker/pull/49

**Comment by tfoote** [comment]
This should be resolved using rocker 0.1.6
